src/Framework/ClassResolver/InstanceCreator/InstanceCreator.php

```diff
@@ -18,14 +18,15 @@ public function __construct(GacelaConfigFileInterface $gacelaConfigFile)
         $this->gacelaConfigFile = $gacelaConfigFile;
     }
 
-    public function createInstance(string $resolvedClassName): ?object
+    public function createByClassName(string $className): ?object
     {
-        if (class_exists($resolvedClassName)) {
+        if (class_exists($className)) {
+            // TODO: Consider caching the dependencies(?)
```

Comment:
(?)

Reply:
It was an idea that I wanted to develop further in a follow up PR.